lib/improver/blending/weights.py

```diff
@@ -285,13 +285,14 @@ def build_weights_cube(cube, weights, blending_coord):
         except ValueError:
             weights_cube = iris.util.new_axis(cube, blending_coord)
             weights_cube = next(weights_cube.slices(blending_coord))
-
         weights_cube.attributes = None
+        # Find dim associated with blending_coord and don't remove any coords
```

Comment:
Agreed - can you give an example of when we were not doing that?  (I don't think we have had issues with losing "time" or "forecast_period" when cycle blending, for example.)

I think this change is fine in principle but am concerned it might have unexpected effects, eg with multi-model blending.

Reply:
At some point in during this ticket I had a problem with anonymous dimensions appearing on my weights cube. I think I still have this problem with one test now (I think it's with model blending not cycle blending). I'm taking a look at it now, and I'll put up the details on the ticket.

Reply:
The problem with this is when we do model blending. Without this change we get a weights cube of:
```
weights: weights / (1)                       (-- : 2)
     Auxiliary coordinates:
          model_configuration           x
     Scalar coordinates:
          forecast_period: 18000 seconds
          forecast_reference_time: 2017-12-08 04:00:00
          time: 2017-12-08 09:00:00
```
and with this change we get
```
weights: weights / (1)                       (model_id: 2)
     Dimension coordinates:
          model_id                           x
     Auxiliary coordinates:
          model_configuration                x
     Scalar coordinates:
          forecast_period: 18000 seconds
          forecast_reference_time: 2017-12-08 04:00:00
          time: 2017-12-08 09:00:00
```
The second case makes it much easier to match coordinates in the weights cube and the cube we are collapsing.
Bats test 12 and 17 and affected by this change and they currently fail without it.

Reply:
OK, it's clear this change is an improvement then - I'm happy with this.